src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +24,50 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
-        }
+        try {
+            ArgumentMultimap argMultimap =
+                    ArgumentTokenizer.tokenize(args, PREFIX_PHONE, PREFIX_EMAIL);
+            FindCommand.FindPersonDescriptor findPersonDescriptor = new FindCommand.FindPersonDescriptor();
+
+            boolean isBothEmpty = !arePrefixesPresent(argMultimap, PREFIX_PHONE)
+                    && !arePrefixesPresent(argMultimap, PREFIX_EMAIL);
+            boolean isBothFilled = arePrefixesPresent(argMultimap, PREFIX_PHONE)
+                    && arePrefixesPresent(argMultimap, PREFIX_EMAIL);
+            if (isBothEmpty) {
+                String trimmedArgs = args.trim();
+                if (trimmedArgs.isEmpty()) {
+                    throw new ParseException(
+                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+                }
+                String[] nameKeywords = trimmedArgs.split("\\s+");
+                return new FindCommand(new AllInfoContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+            } else if (isBothFilled || !argMultimap.getPreamble().isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            } else if (arePrefixesPresent(argMultimap, PREFIX_PHONE)) {
+                Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+                findPersonDescriptor.setPhone(phone);
+            } else if (arePrefixesPresent(argMultimap, PREFIX_EMAIL)) {
+                Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+                findPersonDescriptor.setEmail(email);
+            }
```

Comment:
Perhaps this will need to change to handle multiple similar prefix like "find p/12345678 p/23456789" due to the argMultimap implementation of getValue.

Reply:
Resolved. Thank you!